### PR TITLE
Remove Cloud Endpoints 

### PR DESCRIPTION
**Pull request by ansingh7115**
Main.scala and HammRoutes.scala classes are what you should look at. 
This isn't done yet - just looking for some initial feedback on the strategy.

To Do:
- Swagger page
- Request timeouts
- Add tests back
- Clean up

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/workbench/hamm/Main.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.broadinstitute.workbench.hamm.auth.HttpSamDAO
+import org.broadinstitute.workbench.hamm.dao.{GooglePriceListDAO, WorkflowMetadataDAO}
+import org.broadinstitute.workbench.hamm.service.{StatusService, WorkflowCostService}
+import org.http4s.Uri
+import org.http4s.client.blaze.BlazeClientBuilder
+import org.http4s.server.blaze.BlazeServerBuilder
+
+import scala.concurrent.ExecutionContext.Implicits.global //use better thread pool
+
+object Main extends IOApp with HammLogger {
```
Comment (Qi77Qi) [inline core/src/main/scala/org/broadinstitute/workbench/hamm/Main.scala]: this should live in `server` package...we have a background service depends on `core` as well, and we definitely want to run api server tgt with the background process

File: core/src/main/scala/org/broadinstitute/workbench/hamm/api/HammRoutes.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import org.broadinstitute.workbench.hamm.auth.HttpSamDAO
+import org.broadinstitute.workbench.hamm.model.{HammException, UserInfo, WorkflowId}
+import org.http4s.circe._
+import org.http4s.dsl.Http4sDsl
+import org.http4s._
+import org.http4s.server.middleware.RequestLogger
+import org.http4s.syntax.kleisli._
+import org.http4s.util.CaseInsensitiveString
+
+class HammRoutes(samDAO: HttpSamDAO, workflowCostService: WorkflowCostService, statusService: StatusService)(implicit con: Concurrent[IO]) extends Http4sDsl[IO] with HammLogger {
```
Comment (Qi77Qi) [inline core/src/main/scala/org/broadinstitute/workbench/hamm/api/HammRoutes.scala]: I wouldn't hardcode anything to `IO` except in `Main.scala`

File: core/src/main/scala/org/broadinstitute/workbench/hamm/dao/GooglePriceListDAO.scala [hunk 3]
```diff
@@ -10,24 +10,22 @@ import org.http4s.client.Client
 
 
 
-class GooglePriceListDAO[F[_]: Sync](httpClient: Client[F], uri: Uri) {
+class GooglePriceListDAO(httpClient: Client[IO], uri: Uri) {
 
-  def getGcpPriceList(): F[GooglePriceList] = {
+  def getGcpPriceList(): GooglePriceList = {
```
Comment (Qi77Qi) [inline core/src/main/scala/org/broadinstitute/workbench/hamm/dao/GooglePriceListDAO.scala]: What's the reason removing `F` and making this a blocking IO? Are you planning to rewrapping this in a `IO` at call site?

File: core/src/main/scala/org/broadinstitute/workbench/hamm/dao/WorkflowMetadataDAO.scala [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
+import org.http4s.client.dsl.io._
+import org.http4s.headers._
+import org.http4s.MediaType
+import org.http4s.dsl.io._
+import org.http4s.headers.{Accept, Authorization}
 
-class WorkflowMetadataDAO[F[_]: Sync](httpClient: Client[F], uri: Uri) {
-  def getMetadata(workflowId: WorkflowId): F[MetadataResponse] = {
-    httpClient.expect[MetadataResponse](uri + s"/${workflowId.uuid}/metadata") //TODO: figure out how to get the url properly
+class WorkflowMetadataDAO(httpClient: Client[IO], uri: Uri) {
```
Comment (Qi77Qi) [inline core/src/main/scala/org/broadinstitute/workbench/hamm/dao/WorkflowMetadataDAO.scala]: same comment. Why making everything a blocking IO and throwing up exceptions when error happens?

File: build.sbt [hunk 5]
```diff
@@ -26,38 +18,3 @@ val core =
...
-    .in(file("server"))
-    .enablePlugins(JavaAppPackaging)
-    .settings(
-      libraryDependencies ++= Dependencies.server,
-      Settings.serverSettings
-    )
-    .dependsOn(protobuf)
-    .dependsOn(core % "test->test;compile->compile")
-
-lazy val costUpdater =
```
Comment (Qi77Qi) [inline build.sbt]: we need to keep back ground processor

File: build.sbt [hunk 6]
```diff
@@ -26,38 +18,3 @@ val core =
...
-    .settings(
-      libraryDependencies ++= Dependencies.costUpdater,
-      Settings.costUpdaterSettings
-    )
-    .dependsOn(protobuf)
-    .dependsOn(core % "test->test;compile->compile")
-
-
-lazy val automation =
-  project
```
Comment (Qi77Qi) [inline build.sbt]: we need automation as well

File: core/src/main/scala/org/broadinstitute/workbench/hamm/CostCalculator.scala [hunk 7]
```diff
@@ -11,12 +11,17 @@ import scala.concurrent.duration.FiniteDuration
 
 object CostCalculator {
 
-  def getPriceOfWorkflow(callMetaDataJson: MetadataResponse, priceList: PriceList): Either[Throwable, Double] = {
+  def getPriceOfWorkflow(callMetaDataJson: MetadataResponse, priceList: PriceList): Double = {
```
Comment (Qi77Qi) [inline core/src/main/scala/org/broadinstitute/workbench/hamm/CostCalculator.scala]: What's the benefit of this change?

File: core/src/main/scala/org/broadinstitute/workbench/hamm/auth/HttpSamDAO.scala [hunk 8]
```diff
@@ -1,30 +1,49 @@
...
-  def queryAction(samResource: SamResource, action: String, token: String): F[Boolean] = {
-      httpClient.expect[Boolean](
-        Request[F](
-          Method.GET,
-          Uri.unsafeFromString(uri + s"/api/resources/v1/workflow-collection/${samResource.resourceName}/action/$action"),
-          headers = Headers(Header("authorization", token))))
+  def queryAction(token: String, samResource: SamResource, action: String): Boolean = {
+    val url = uri + s"/api/resources/v1/workflow-collection/${samResource.resourceName}/action/$action"
+    val request = GET(uri = Uri.unsafeFromString(url), Authorization(Credentials.Token(AuthScheme.Bearer, token)), Accept(MediaType.application.json))
+    httpClient.expect[Boolean](request).unsafeRunSync()
```
Comment (Qi77Qi) [inline core/src/main/scala/org/broadinstitute/workbench/hamm/auth/HttpSamDAO.scala]: in general, `.unsafeRunSync()` should really only happen in unit test..it's quite rare you need to `unsafeRunSync()` in app code

File: core/src/main/resources/org/broadinstitute/dsde/workbench/hamm/liquibase/changesets/20190225_create_table_workflow.xml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+            <column name="ID" type="SERIAL"></column>
+            <column name="WORKFLOW_ID" type="VARCHAR(255)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="PARENT_WORKFLOW_ID" type="VARCHAR(255)"></column>
+            <column name="ROOT_WORKFLOW_ID" type="VARCHAR(255)"></column>
+            <column name="WORKFLOW_COLLECTION_ID" type="VARCHAR(255)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="IS_SUBWORKFLOW" type="BOOLEAN">
```
Comment (dvoet) [inline core/src/main/resources/org/broadinstitute/dsde/workbench/hamm/liquibase/changesets/20190225_create_table_workflow.xml]: do we need this in addition to `PARENT_WORKFLOW_ID`

File: core/src/main/resources/org/broadinstitute/dsde/workbench/hamm/liquibase/changelog.xml [hunk 10]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+    <include file="changesets/20190225_create_table_job.xml" relativeToChangelogFile="true"/>
+    <include file="changesets/20190225_create_table_workflow.xml" relativeToChangelogFile="true"/>
+    <include file="changesets/20190225_job_unique_key.xml" relativeToChangelogFile="true"/>
+    <include file="changesets/20190225_workflow_label_index.xml" relativeToChangelogFile="true"/>
```
Comment (dvoet) [inline core/src/main/resources/org/broadinstitute/dsde/workbench/hamm/liquibase/changelog.xml]: you don't need a separate file for each change set

**Pull request merged**